packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumPlugin.kt

```diff
@@ -75,32 +91,29 @@ class DatadogRumPlugin(
         channel.setMethodCallHandler(null)
     }
 
-    fun attachToExistingSdk() {
-        rum = GlobalRum.get()
-    }
-
-    fun setup(
-        configuration: DatadogFlutterConfiguration.RumConfiguration
-    ) {
-        rum = RumMonitor.Builder()
-            .sampleRumSessions(configuration.sampleRate)
-            .build()
-        GlobalRum.registerIfAbsent(rum!!)
-    }
+//    fun attachToExistingSdk() {
+//        rum = GlobalRum.get()
+//    }
 
     override fun onMethodCall(call: MethodCall, result: Result) {
+        if (call.method != "enable" && rum == null) {
+            result.invalidOperation("Attempting to use RUM when it has not been enabled")
+            return
+        }
+
         try {
             when (call.method) {
+                "enable" -> enable(call, result)
                 "startView" -> startView(call, result)
                 "stopView" -> stopView(call, result)
                 "addTiming" -> addTiming(call, result)
-                "startResourceLoading" -> startResourceLoading(call, result)
-                "stopResourceLoading" -> stopResourceLoading(call, result)
-                "stopResourceLoadingWithError" -> stopResourceLoadingWithError(call, result)
+                "startResource" -> startResourceLoading(call, result)
+                "stopResource" -> stopResourceLoading(call, result)
+                "stopResourceWithError" -> stopResourceLoadingWithError(call, result)
                 "addError" -> addError(call, result)
-                "addUserAction" -> addUserAction(call, result)
-                "startUserAction" -> startUserAction(call, result)
-                "stopUserAction" -> stopUserAction(call, result)
+                "addAction" -> addUserAction(call, result)
+                "startAction" -> startUserAction(call, result)
+                "stopAction" -> stopUserAction(call, result)
```

Comment:
minor: should we rename internal methods as well just to align?

Reply:
Yeah that's a good call.